lua/telescope/actions/init.lua

```diff
@@ -225,6 +225,49 @@ actions.run_builtin = function(prompt_bufnr)
   require('telescope.builtin')[entry.text]()
 end
 
+-- Maybe upstream in plenary. TJ was already working on a new Path module
+-- Or maybe tami5 is interested.
+local create_file = function(file)
+  vim.loop.fs_open(file, "w", 420, vim.schedule_wrap(function(err, fd)
+    if err then
+      vim.api.nvim_err_writeln('Could not create file: ' .. file)
+    else
+      vim.loop.fs_close(fd)
+    end
+  end))
+end
+
+actions.create_file = function(prompt_bufnr)
+  local file = vim.fn.input('Create file or directory(add / at the end of file). Enter name: ')
+  if file == "" then return end
+
+  local fpath = actions.get_selected_entry(prompt_bufnr).path
+  fpath = string.sub(fpath, -1) == '/' and vim.fn.fnamemodify(fpath, ':h:h') or vim.fn.fnamemodify(fpath, ':h')
+  fpath = fpath .. '/' .. file
+
+  actions.close(prompt_bufnr)
+
+  if string.sub(fpath, -1) ~= '/' then
+    create_file(fpath)
+    vim.cmd(string.format(':e %s', fpath))
+  else
+    -- TODO(conni2461): OHHHHH, We could do that recursive. Like enabling doing something like this:
+    -- path/to/test/dir/ or even creating a file at the end
+    -- path/to/test/file.lua
+    -- Would be cool.
+    -- Also if dir is last entry. Open file_browser in last dir
+    -- Or if file is last entry. Open file
+    local fd = vim.loop.fs_mkdir(fpath, 493)
+    if not fd then
+      vim.api.nvim_err_writeln('Could not create directory: ' .. fpath)
+      return
+    end
+    -- TODO(conni2461): We should keep track of the opts. We don't have them here right? So we keep opts.show_hidden
+    require'telescope.builtin'.file_browser({ cwd = fpath })
```

Comment:
we could save the opts for the picker if we wanted, the same way we do the for the last selected entry.